### DISCUSSION TITLE
tailwindcss watch mode doesn't work in bash script

### DISCUSSION THREAD

**Discussion by elgs** [tailwindlabs/tailwindcss#9818] [Help]
**What version of Tailwind CSS are you using?**

tailwindcss v3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

None

**What version of Node.js are you using?**

N/A

Downloaded from 
`curl -sLO https://github.com/tailwindlabs/tailwindcss/releases/latest/download/tailwindcss-macos-arm64`

**What browser are you using?**

N/A

**What operating system are you using?**

macOS

**Reproduction URL**

N/A

**Describe your issue**

My `dev.sh` looks like this. When I run `./dev.sh`, the app.css wouldn't be created.
```bash
#!/usr/bin/env bash
tailwindcss -i src/tailwind.css -o dist/app.css -w &
```

However, any of the following works:
```bash
#!/usr/bin/env bash

# removed the ampersand
tailwindcss -i src/tailwind.css -o dist/app.css -w
```

```bash
#!/usr/bin/env bash
# but doesn't watch
tailwindcss -i src/tailwind.css -o dist/app.css &
```

or run it directly:
```bash
$ tailwindcss -i src/tailwind.css -o dist/app.css -w &
```

**Comment by thecrypticace** [comment]
This happens because, in a bash script, pushing a job to the background closes stdin because there's no longer a pipe for it to be connected to. We use closing stdin as a signal that the process needs to exit (using process exit signals is not 100% reliable and can result in processes sticking around indefinitely).

I unfortunately do not have a workaround but the general idea, I think, is to make sure that stdin still has a file descriptor attached even after moving to the background. I'm not sure how to actually achieve this though. Sorry.

**Comment by elgs** [comment]
Thanks for the insight @thecrypticace.

I just tried and none of the following worked:
```bash
 #!/usr/bin/env bash
tailwindcss -i src/tailwind.css -o dist/app.css -w </dev/null & 
```

```bash
 #!/usr/bin/env bash
touch a
tailwindcss -i src/tailwind.css -o dist/app.css -w <./a & 
```

> This happens because, in a bash script, pushing a job to the background closes stdin because there's no longer a pipe for it to be connected to. We use closing stdin as a signal that the process needs to exit (using process exit signals is not 100% reliable and can result in processes sticking around indefinitely).

I am not sure if this is a good idea or not. This is effectively making it impossible to run tailwindcss watch mode in a script.

**Comment by stfnx** [comment]
This happens only in recent versions and worked before: #9870

**Comment by charlesdeb** [comment]
Hi folks, I am having a similar issue. I want to use multiple tailwind configurations and have them both being watched as I develop. In my `package.json`, this doesn't work:

```
  "scripts": {
    "watch": "wp-scripts start & (npx tailwindcss -i ./src/css/wp_main.css -o ./dist/css/wp_main.css  --watch) & (npx tailwindcss -i ./src/css/admin_main.css -o ./dist/css/admin_main.css --watch)",
....
}
```
Before i added the second `tailwindcss`  command it used to work. But with both of them, only one of the `tailwindcss` scripts starts up with a `Rebuilding...` message to the command line. Removing the `--watch` causes both `tailwindcss` processes to run - but that means I need to manually run it on every code change. So, not really idea.

Just running `(npx tailwindcss -i ./src/css/wp_main.css -o ./dist/css/wp_main.css  --watch) & (npx tailwindcss -i ./src/css/admin_main.css -o ./dist/css/admin_main.css --watch)` on the command line works at first, but after a few seconds I get another error:

```
Error: ENOSPC: System limit for number of file watchers reached, watch '/path/to/some/file/in/my/project/
...
...
...
other node error handling gubbins
```
So, I'm not too sure what to do about this. Suggestions?

**Comment by mmorris8** [comment]
reasonsTailWindSucks++

Ran across this today. Oh well - I'm working on getting rid of all Tailwind in the project I'm working on.  It's not a bad framework for rapid prototyping I guess, but for ongoing projects it an utterly boneheaded approach.  Taking style declarations out of the HTML was the REASON CSS was created - and all Tailwind accomplishes is putting them back in. 1998 called, and it wants its styling system back.
